client/client_impl.go

```diff
@@ -71,18 +73,31 @@ func NewClientToken(uri, token string) Client {
 
 // NewClientTokenTLS returns a client at the specified url that authenticates
 // all outbound requests with the given token and tls.Config if provided.
-func NewClientTokenTLS(uri, token string, c *tls.Config) Client {
+func NewClientTokenTLS(uri, token string, c *tls.Config) (Client, error) {
 	config := new(oauth2.Config)
 	auther := config.Client(oauth2.NoContext, &oauth2.Token{AccessToken: token})
 	if c != nil {
 		if trans, ok := auther.Transport.(*oauth2.Transport); ok {
-			trans.Base = &http.Transport{
-				TLSClientConfig: c,
-				Proxy:           http.ProxyFromEnvironment,
+			if os.Getenv("SOCKS_PROXY") != "" {
+				dialer, err := proxy.SOCKS5("tcp", os.Getenv("SOCKS_PROXY"), nil, proxy.Direct)
+				if err != nil {
+					fmt.Fprintln(os.Stderr, "can't connect to the proxy:", err)
```

Comment:
this printlin can be removed. The parent function that invokes and handles the error will be responsible for logging the error message